extensions/ql-vscode/src/variant-analysis/view-autofixes.ts

```diff
@@ -0,0 +1,886 @@
+            variantAnalysisIdStoragePath,
+            nwo,
+          );
+
+          // Read the contents of the variant analysis' `repo_task.json` file,
+          // and confirm that the `databaseCommitSha` and `resultCount` exist.
+          const repoTask: VariantAnalysisRepositoryTask = await readRepoTask(
+            variantAnalysisResultsManager.getRepoStorageDirectory(
+              variantAnalysisIdStoragePath,
+              nwo,
+            ),
+          );
+          if (!repoTask.databaseCommitSha) {
+            throw new Error(`Missing database commit SHA for ${nwo}`);
+          }
+          if (!repoTask.resultCount) {
+            throw new Error(`Missing variant analysis result count for ${nwo}`);
+          }
+
+          // Download the source root.
+          // Using `0` as the progress step to force a dynamic vs static progress bar.
+          progressForRepo(progressUpdate(0, 3, `Fetching source root...`));
+          const srcRootPath = await downloadPublicCommitSource(
+            nwo,
+            repoTask.databaseCommitSha,
+            sourceRootsStoragePath,
+            credentials,
+            progressForRepo,
+          );
+
+          // Run autofix.
+          progressForRepo(progressUpdate(2, 3, `Running autofix...`));
+          await runAutofixForRepository(
+            nwo,
+            sarifFile,
+            srcRootPath,
+            localAutofixPath,
+            autofixOutputStoragePath,
+            repoTask.resultCount,
+            outputTextFiles,
+          );
+        },
+        {
+          title: `${nwo}`,
+          cancellable: false,
+        },
+      ),
+    ),
+  );
+
+  return outputTextFiles;
+}
+
+/**
+ * Gets the path to a SARIF file for a given `nwo`.
+ */
+async function getRepoSarifFile(
+  variantAnalysisResultsManager: VariantAnalysisResultsManager,
+  variantAnalysisIdStoragePath: string,
+  nwo: string,
+): Promise<string> {
+  if (
+    !(await variantAnalysisResultsManager.isVariantAnalysisRepoDownloaded(
+      variantAnalysisIdStoragePath,
+      nwo,
+    ))
+  ) {
+    throw new Error(`Variant analysis results not downloaded for ${nwo}`);
+  }
+  const sarifFile =
+    variantAnalysisResultsManager.getRepoResultsSarifStoragePath(
+      variantAnalysisIdStoragePath,
+      nwo,
+    );
+  if (!(await pathExists(sarifFile))) {
+    throw new Error(`SARIF file not found for ${nwo}`);
+  }
+  return sarifFile;
+}
+
+/**
+ * Downloads the source code of a public commit from a GitHub repository.
+ */
+async function downloadPublicCommitSource(
+  nwo: string,
+  sha: string,
+  outputPath: string,
+  credentials: Credentials,
+  progressCallback: ProgressCallback,
+): Promise<string> {
+  const [owner, repo] = nwo.split("/");
+  if (!owner || !repo) {
+    throw new Error(`Invalid repository name: ${nwo}`);
+  }
+
+  // Create output directory if it doesn't exist
+  await ensureDir(outputPath);
+
+  // Define the final checkout directory
+  const checkoutDir = join(
```

Comment:
I think I'm missing something, but where do we define that the archive is extracted to this directory?

Reply:
Ah, thanks for catching this. This got affected by my refactor. The `checkoutDir` here ends up being the same directory name that `unzipToDirConcurrently` extracts to so it was still working. Fixed in https://github.com/github/vscode-codeql/pull/4065/commits/cb4bfbc517accded65ee458452cec61257a5ce0a.